src/select/select.js

```diff
@@ -69,168 +107,126 @@ class Select extends React.Component<PropsT, SelectStateT> {
-      if (__BROWSER__) {
+    if (__BROWSER__) {
+      if (prevState.isOpen !== this.state.isOpen) {
+        if (this.state.isOpen) {
+          this.props.onOpen && this.props.onOpen();
+          document.addEventListener('touchstart', this.handleTouchOutside);
+        } else {
+          this.props.onClose && this.props.onClose();
+          document.removeEventListener('touchstart', this.handleTouchOutside);
+        }
+      }
+
+      if (!prevState.isFocused && this.state.isFocused) {
         document.addEventListener('click', this.handleClickOutside);
       }
     }
   }
 
   componentWillUnmount() {
-    this.toggleTouchOutsideEvent(false);
-  }
-
-  toggleTouchOutsideEvent(enabled: boolean) {
     if (__BROWSER__) {
-      if (enabled) {
-        document.addEventListener('touchstart', this.handleTouchOutside);
-      } else {
-        document.removeEventListener('touchstart', this.handleTouchOutside);
-      }
+      document.removeEventListener('touchstart', this.handleTouchOutside);
     }
   }
 
-  handleTouchOutside = (event: Event) => {
-    // Handle touch outside on ios to dismiss menu
-    // $FlowFixMe
-    if (this.wrapper && !this.wrapper.contains(event.target)) {
-      this.closeMenu();
-    }
-  };
-
   focus() {
     if (!this.input) return;
     this.input.focus();
   }
 
-  handleTouchMove = () => {
-    // Set a flag that the view is being dragged
-    this.dragging = true;
-  };
-
-  handleTouchStart = () => {
-    // Set a flag that the view is not being dragged
-    this.dragging = false;
+  // Handle touch outside on mobile to dismiss menu, ensures that the
+  // touch target is not within the anchor DOM node.
+  handleTouchOutside = (event: TouchEvent) => {
+    if (!containsNode(this.anchor.current, event.target)) {
+      this.closeMenu();
+    }
   };
 
-  handleTouchEnd = (event: Event) => {
-    // Check if the view is being dragged. In this case
-    // we don't want to fire the click event (because the user only wants to scroll)
+  // Track dragging state to filter false-positive actions where a user
+  // intends to drag/scroll the page.
+  handleTouchMove = () => (this.dragging = true);
+  handleTouchStart = () => (this.dragging = false);
+  handleTouchEnd = (event: TouchEvent) => {
     if (this.dragging) return;
-    // Fire the mouse events
     this.handleClick(event);
   };
-
-  handleTouchEndClearValue = (event: Event) => {
-    // Check if the view is being dragged. In this case
-    // we don't want to fire the click event (because the user only wants to scroll)
+  handleTouchEndClearValue = (event: TouchEvent) => {
     if (this.dragging) return;
-    // Clear the value
     this.clearValue(event);
   };
 
-  handleClickOutside = (event: Event) => {
-    if (this.isValueJustSelected) {
-      this.isValueJustSelected = false;
+  handleClick = (event: MouseEvent | TouchEvent) => {
+    if (this.props.disabled || !isClick(event) || !isLeftClick(event)) {
```

Comment:
It should probably still be `if (this.props.disabled || (!isClick(event) && !isLeftClick(event))) {`

Reply:
```suggestion
    if (this.props.disabled || (!isClick(event) && !isLeftClick(event))) {
```